R/reporter-junit.R

```diff
@@ -0,0 +1,153 @@
+#' @include reporter.R
+NULL
+
+classnameOK <- function(text) {
+  gsub("[ \\.]", "_", text)
+}
+
+add_broken <- function (parent, type, message) {
```

Comment:
Don't need assignment any more, and maybe should be `add_failure_node()`?  Are you sure the `type` is both the node name and an attribute?